verify/cert.go

```diff
@@ -155,7 +176,10 @@ func (cv *CertVerifier) getConfDeepBlockNumber(ctx context.Context) (*big.Int, e
 }
 
 // retrieveBatchMetadataHash retrieves the batch metadata hash stored on-chain at a specific blockNumber for a given batchID
-// returns an error if some problem calling the contract happens, or the hash is not found
+// returns an error if some problem calling the contract happens, or the hash is not found.
+// We make an eth_call to the EigenDAServiceManager at the given blockNumber to retrieve the hash.
+// Therefore, make sure that blockNumber is <128 blocks behind the latest block, to avoid requiring an archive node.
+// This is currently enforced by having EthConfirmationDepth be <64.
```

Comment:
Should we poll the latest block number and ensure it's greater than the confirmation block number?

Reply:
i.e, do we care about the case where the RPC node is far behind?

Reply:
Good question. I personally don't think so. For 3 reasons:
1. what would we do anyways? sleep until that block catches up? Don't think this is a valid strategy. Think its better to return an error and let the outer service (batcher/node) retry. This will also raise alerts and get the human in the loop to intervene
2. sequencer should have an up to date eth node as a requirement
3. validators will typically be many many blocks ahead, so their RPC would have to be suuuuuper stale anyways for this case to happen

Reply:
at a minimum we could fail loudly within proxy indicating that the node has a stale state. agree though its a performance degradation (i.e, additional rpc query) in-favor of better devx.

Reply:
Don't think we should fail loudly. It might just be a temporary glitch. That's why retries exist. Failure should happen after some number of retries from the outer loop service.